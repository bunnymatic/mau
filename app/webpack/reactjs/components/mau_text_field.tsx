import { FieldError } from "@reactjs/components/field_error";
import { MauHint } from "@reactjs/components/mau_hint";
import { ErrorMessage, Field } from "formik";
import React, { FC, JSX } from "react";

interface MauTextFieldProps {
  id?: string;
  name: string;
  label: string;
  placeholder: string;
  className?: string;
  hint?: string | JSX.Element;
}

export const MauTextField: FC<MauTextFieldProps> = ({
  id,
  name,
  label,
  placeholder,
  className,
  hint,
}) => {
  const classes = className ?? "";
  return (
    <div className={`string input optional ${classes}`}>
      <label htmlFor={name}>{label}</label>
      <Field
        type="text"
        name={name}
        placeholder={placeholder}
        id={id || name}
      />
      <ErrorMessage component={FieldError} name={name} />
      <MauHint>{hint}</MauHint>
    </div>
  );
};
