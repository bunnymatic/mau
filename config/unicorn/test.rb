# frozen_string_literal: true

# Bind to 127.0.0.1 to avoid issue with MacOS and `getaddrinfo`
# causing ruby crashes during asset compilation and startup.
# https://bugs.ruby-lang.org/issues/15763
listen '127.0.0.1:3000'
