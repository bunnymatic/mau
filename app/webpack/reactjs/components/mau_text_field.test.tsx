import { renderInForm } from "@reactjs/test/renderers";
import React from "react";
import { describe, expect, it } from "vitest";

import { MauTextField } from "./mau_text_field";

describe("MauTextField", () => {
  it("matches the snapshot", () => {
    const { container } = renderInForm(
      <MauTextField
        label="The Label"
        name="the_name"
        placeholder="e.g. like this"
        hint={<span>here is a clue</span>}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot if id is provided", () => {
    const { container } = renderInForm(
      <MauTextField
        id="different-than-name"
        label="The Label"
        name="the_name"
        placeholder="e.g. like this"
        hint={<span>here is a clue</span>}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot if type is provided", () => {
    const { container } = renderInForm(
      <MauTextField
        type="email"
        label="The Label"
        name="the_name"
        placeholder="e.g. email"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
