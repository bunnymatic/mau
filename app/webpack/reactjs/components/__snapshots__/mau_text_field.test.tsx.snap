// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MauTextField > matches the snapshot 1`] = `
<div>
  <div
    class="string input optional "
  >
    <label
      for="the_name"
    >
      The Label
    </label>
    <input
      id="the_name"
      name="the_name"
      placeholder="e.g. like this"
      type="text"
      value=""
    />
    <p
      class="inline-hints"
    >
      <span>
        here is a clue
      </span>
    </p>
  </div>
</div>
`;

exports[`MauTextField > matches the snapshot if id is provided 1`] = `
<div>
  <div
    class="string input optional "
  >
    <label
      for="the_name"
    >
      The Label
    </label>
    <input
      id="different-than-name"
      name="the_name"
      placeholder="e.g. like this"
      type="text"
      value=""
    />
    <p
      class="inline-hints"
    >
      <span>
        here is a clue
      </span>
    </p>
  </div>
</div>
`;

exports[`MauTextField > matches the snapshot if type is provided 1`] = `
<div>
  <div
    class="string input optional "
  >
    <label
      for="the_name"
    >
      The Label
    </label>
    <input
      id="the_name"
      name="the_name"
      placeholder="e.g. email"
      type="email"
      value=""
    />
  </div>
</div>
`;
